project_status.py

```diff
@@ -298,25 +295,355 @@ def main(args):
                     sample_entry['MedianInsertSize']
                 )
 
+def get_low_coverage(project):
+    # run same script again, to parse the output
+    # because all the functions above don't return anything, but just print the result
+    script_path = os.path.realpath(__file__)
+    command = """python {} {} | awk -v x=28.5 '$7<x' | awk '{{ print $1 " " $7}}' """.format(script_path, project)
+    try:
+        p = subprocess.Popen(command, shell=True, stdout=subprocess.PIPE, stderr=subprocess.PIPE, universal_newlines=True)
+    except Exception, e:
+        print "Command failed: {}".format(command)
+        raise e
+    output = p.communicate()[0]
+
+    result = {}
+    for line in output.split('\n'):
+        try:
+            sample, coverage = line.split()
+        except:
+            continue
+        else:
+            result[sample] = coverage
+    return result
+
+def get_project_status(project):
+    # run same script again, to parse the output
+    # because all the functions above don't return anything, but just print the result
+    script_path = os.path.realpath(__file__)
+    command = """python {} {} --project-status --skip-header""".format(script_path, project)
+    try:
+        p = subprocess.Popen(command.split(), stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+    except Exception, e:
+        print "Command failed: {}".format(command)
+        raise e
+    output, error = p.communicate()
+    if error:
+        print error
+        exit(1)
+
+    result = {
+        'sequenced': [],
+        'resequenced': [],
+        'organized': [],
+        'not_organized': [],
+        'analyzed': [],
+        'to_analyze': [],
+    }
+
+    for line in output.split('\n'):
+        try:
+            sample, sequenced, organized, analysed = line.split()
+            if project not in sample:
+                continue
+        except Exception,e: continue
+        result['sequenced'].append(sample)
+        if sequenced > '1':
+            result['resequenced'].append(sample)
+        if organized != '0':
+            result['organized'].append(sample)
+        if organized < sequenced:
+            result['not_organized'].append(sample)
+        if analysed != '0':
+            result['analyzed'].append(sample)
+        if analysed < sequenced:
+            result['to_analyze'].append(sample)
+    return result
+
+def get_samples_with_undetermined(data_dir, project):
+    """ get all fastq_files from DATA directory
+        check which ones named 'Undetermined'
+        then add sample and flowcell to the list
+    """
+    result = {}
+    # get list of fastq_files in DATA directory
+    project_path = os.path.join(data_dir, project, '*/*/*/*.fastq*')
```

Comment:
you can make some more strict assumption:
 - Demultiplexing folder
 - data is always fastq.gz